diff --git a/dist/index.d.ts b/dist/index.d.ts
index ddb2634f102ce931beb4d2285101b738b9afa888..374bc8a85491d00fb0cedb8e97fd128e3a2a5263 100644
--- a/dist/index.d.ts
+++ b/dist/index.d.ts
@@ -1,5 +1,7 @@
 import type { Compiler, RspackPluginInstance } from "@rspack/core";
 export interface IPreactRefreshRspackPluginOptions {
+    include?: string | RegExp | (string | RegExp)[] | null;
+    exclude?: string | RegExp | (string | RegExp)[] | null;
     overlay?: {
         module: string;
     };
diff --git a/dist/index.js b/dist/index.js
index 0a153309db2f0919049e91af0a52d303544ab285..8a6bb2ae125c07e574d1640c6713fb7d0067516e 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -46,6 +46,8 @@ class PreactRefreshRspackPlugin {
         this.options = options;
         this.name = NAME;
         this.options = {
+            include: options?.include ?? /\.([jt]sx?)$/,
+            exclude: options?.exclude ?? /node_modules/,
             overlay: options?.overlay
         };
     }
@@ -72,9 +74,9 @@ class PreactRefreshRspackPlugin {
             ...compiler.options.resolve.alias
         };
         compiler.options.module.rules.unshift({
-            include: /\.([jt]sx?)$/,
+            include: this.options.include,
             exclude: {
-                or: [/node_modules/, ...INTERNAL_PATHS].filter(Boolean)
+                or: [this.options.exclude, ...INTERNAL_PATHS].filter(Boolean)
             },
             use: "builtin:preact-refresh-loader"
         });
